lambdas/file_access_patterns/lambda_for_fileDelete/index.js

```diff
@@ -8,97 +8,92 @@ function response(statusCode, message) {
-  }
-  catch(err){
-    return response(err.statusCode,err.message)
+  try {
+    var fileData = await dynamo.delete(params).promise();
+  } catch (err) {
+    return response(err.statusCode, err.message);
   }
 
   //updating the storage_used for the user who deleted the file
-  try{
-    await dynamo.update({
-      TableName: "V-Transfer",
-      Key: {
-        PK: `USER#${event.path.u_id}`,
-        SK: `METADATA`, 
-      },
-      UpdateExpression: "add storage_used :size",
-      ExpressionAttributeValues: {
-        ":size": -fileData.Attributes.size,
-      },
-    }).promise()
+  try {
+    await dynamo
+      .update({
+        TableName: "V-Transfer",
+        Key: {
+          PK: `USER#${event.path.userId}`,
+          SK: `METADATA`,
+        },
+        UpdateExpression: "add storage_used :size",
+        ExpressionAttributeValues: {
+          ":size": -fileData.Attributes.size,
+        },
+      })
+      .promise();
+  } catch (err) {
+    return response(err.statusCode, err.message);
   }
-  catch(err){
-    return response(err.statusCode,err.message)
-  }
-  
-  params={
-    TableName:"V-Transfer",
+
+  params = {
+    TableName: "V-Transfer",
     KeyConditionExpression: "#PK= :pk and begins_with(#SK,:sk)",
-    ExpressionAttributeNames:{
-        "#PK": "PK",
-        "#SK": "SK"
+    ExpressionAttributeNames: {
+      "#PK": "PK",
+      "#SK": "SK",
     },
-    ExpressionAttributeValues:{
-        ':pk':`FILE#${reqBody.f_timestamp}`,
-        ':sk':"URL#"
-    },                        
-  }
+    ExpressionAttributeValues: {
+      ":pk": `FILE#${event.path.fileId}`,
+      ":sk": "URL#",
+    },
+  };
 
   //getting all URLs corresponding to the deleted file record
-  try{
-    var url_data = await dynamo.query(params).promise()        
-    url_data=url_data.Items;
-    if(url_data.length==0){
-      return response(201,"File delete success")
+  try {
+    var url_data = await dynamo.query(params).promise();
+    url_data = url_data.Items;
+    if (url_data.length == 0) {
+      return response(201, "File delete success");
     }
-  }
-  catch(err){
-    return response(err.statusCode,err.message)
+  } catch (err) {
+    return response(err.statusCode, err.message);
   }
 
-  url_data.forEach(request_item_arr_util)
-  params={
-    "RequestItems":{
-      "V-Transfer":request_item_arr
-    }
-  }
-  
+  url_data.forEach(request_item_arr_util);
+  params = {
+    RequestItems: {
+      "V-Transfer": request_item_arr,
+    },
+  };
+
   //deleting all URL records corresponding to the deleted file record
-  try{
-    await dynamo.batchWrite(params).promise()
-    return response(201,"File delete success")
-  }
-  catch(err){
-    return response(err.statusCode,err.message)
+  try {
+    await dynamo.batchWrite(params).promise();
+    return response(201, "File delete success");
+  } catch (err) {
+    return response(err.statusCode, err.message);
   }
-
-}
-
+};
```

Comment:
for this lambda, you can check out this article [How to delete huge data from DynamoDB table](https://medium.com/analytics-vidhya/how-to-delete-huge-data-from-dynamodb-table-f3be586c011c)

Reply:
The current lambda is working fine is there anything to be updated?

Reply:
read the article if you think you are doing the it think then its fine else, suggest changes